packages/common/src/utils/NumberUtil.ts

```diff
@@ -0,0 +1,49 @@
+import BigNumber from 'bignumber.js'
+
+export const NumberUtil = {
+  bigNumber(value: BigNumber.Value) {
+    return new BigNumber(value)
+  },
+
+  /**
+   * Format the given number or string to human readable numbers with the given number of decimals
+   * @param value - The value to format. It could be a number or string. If it's a string, it will be parsed to a float then formatted.
+   * @param decimals - number of decimals after dot
+   * @returns
+   */
+  formatNumberToLocalString(value: string | number | undefined, decimals = 2) {
+    if (value === undefined) {
+      return '0.00'
+    }
+
+    if (typeof value === 'number') {
+      return value.toLocaleString('en-US', {
+        maximumFractionDigits: decimals,
+        minimumFractionDigits: decimals
+      })
+    }
+
+    return parseFloat(value).toLocaleString('en-US', {
+      maximumFractionDigits: decimals,
+      minimumFractionDigits: decimals
+    })
+  },
+
+  /**
+   * Multiply two numbers represented as strings with BigNumber to handle decimals correctly
+   * @param a string
+   * @param b string
+   * @returns
+   */
+  multiply(a: BigNumber.Value | undefined, b: BigNumber.Value | undefined) {
+    if (a === undefined || b === undefined) {
+      // eslint-disable-next-line new-cap
+      return BigNumber(0)
+    }
+
+    const aBigNumber = new BigNumber(a)
+    const bBigNumber = new BigNumber(b)
+
+    return aBigNumber.multipliedBy(bBigNumber)
+  }
```

Comment:
While calculating numbers with too small values, we might need to show the user like `0.000..0634`, etc. This is not quite possible to handle with JS's number utilities. bignumber.js allows us to make calculations with both strings and numbers and return the valid equivalents.

It might have a downside of having another dependency and bundle size, but at the same time, IMO trying to handle numbers manually might tent to have wrong outputs.

One thing here I want to double check that, in none of the UIs, we don't show really long tails of the number. This means that we 'might' not need this actually. I'll double check this